apps/fdsnws/test/test-station-conditional.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+
+from __future__ import absolute_import, division, print_function
+
+import sys
+import time
+
+from twisted.web.http import datetimeToString, stringToDatetime
+
+from station import TestStationBase
+
+
+###############################################################################
+class TestStationConditionalRequests(TestStationBase):
+
+    def command(self):
+        cmd = super(TestStationConditionalRequests, self).command()
+        cmd.append('--handleConditionalRequests=true')
+        return cmd
+
+    def test(self):
+        def validateDateString(dateString):
+            try:
+                _ = stringToDatetime(dateString)
+            except ValueError as e:
+                raise ValueError(
+                    'Invalid date string: {!r}'.format(dateString))
+            return True
+
+        def createQueryString(respFormat, level, net=None, sta=None, loc=None,
+                              cha=None):
+            d = {'level': level, 'format': respFormat}
+            if net:
+                d['network'] = net
+            if sta:
+                d['station'] = sta
+            if loc:
+                d['location'] = loc
+            if cha:
+                d['channel'] = cha
+
+            return '{}?{}'.format(
+                self.query(), '&'.join(
+                    '{}={}'.format(
+                        k, v) for k, v in d.items()))
+
+        def createQueryStrings(respFormat, levels, **kwargs):
+            return [createQueryString(respFormat, l, **kwargs) for l in levels]
+
+        def createTestCases(respFormat, levels, ifModifiedSince,
+                            contentType=None, dataFile=None, retCode=200,
+                            respHeaders=None, **kwargs):
+            return [(qs, contentType, retCode,
+                     {'If-Modified-Since': datetimeToString(ifModifiedSince)},
+                     respHeaders, dataFile)
+                    for qs in createQueryStrings(respFormat, levels,
+                                                 **kwargs)]
+
+        print('Testing station service (conditional requests)')
+
+        epoch = time.mktime(time.gmtime(0))
```

Comment:
@gempa-stephan, you were totally right - my time is set to UTC.

Reply:
Though, the actual issue most probably is located somewhere else. That is, `Wed, 31 Dec 1969 23:00:00 GMT` is according to [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.1.1) a valid HTTP-date. Thus, the implementation must handle it correctly, and therefore the `Last-Modified` header field must be sent along with the HTTP status code **200** response.

- Could you check the value of the `self._timeInventoryLoaded` member variable in `class FDSNStation`?
- Also we need to make sure that the `_isConditionalRequest` method returns `True` in case of HTTP-dates < *epoch*.